src/shogun/CMakeLists.txt

```diff
@@ -12,6 +12,16 @@ include(CheckCXXSourceCompiles)
 include(CMakePackageConfigHelpers)
 
 FILE(GLOB_RECURSE LIBSHOGUN_SRC *.${EXT_SRC_CPP} *.${EXT_SRC_C})
+FOREACH(src ${LIBSHOGUN_SRC})
```

Comment:
@vigsterkr that is a local hack I am using to enforce compiling `SGObject.cpp` happens as first CXX. Very useful to get compiler errors quickly when playing with `SGObject.h`, as otherwise I always need to wait for a bunch of other classes to be compiled first. Not sure something like this should be in, but I put it here for discussion anyways.

Reply:
@karlnapf if you would use ninja then you could call the specific object build.... this is a hack that nobody else needs and your problem could be solved by using ninja.